osu.Game/Screens/Edit/Timing/TimingScreen.cs

```diff
@@ -132,6 +133,61 @@ protected override void LoadComplete()
                 }, true);
             }
 
+            protected override void Update()
+            {
+                base.Update();
+
+                trackActivePoint();
+            }
+
+            /// <summary>
+            /// Given the user has selected a control point group, we want to track any group which is
+            /// active at the current point in time which matches the type the user has selected.
+            ///
+            /// So if the user is currently looking at a timing point and seeks into the future, a
+            /// future timing point would be automatically selected if it is now the new "current" point.
+            /// </summary>
+            private void trackActivePoint()
+            {
+                // For simplicity only match on the first type of the active control point.
+                var selectedPointType = selectedGroup.Value?.ControlPoints.FirstOrDefault()?.GetType();
+
+                if (selectedPointType != null)
+                {
+                    // We don't have an efficient way of looking up groups currently, only individual point types.
+                    // To improve the efficiency of this in the future, we should reconsider the overall structure of ControlPointInfo.
+                    IEnumerable<ControlPointGroup> groups = Beatmap.ControlPointInfo.Groups;
+
+                    bool currentTimeBeforeSelectedGroup = clock.CurrentTimeAccurate < selectedGroup.Value.Time;
+
+                    // Decide whether we are searching backwards or forwards.
+                    if (currentTimeBeforeSelectedGroup)
+                        groups = groups.Reverse();
+
+                    // Find the next group which has the same type as the selected one.
+                    groups = groups.SkipWhile(g => g != selectedGroup.Value)
+                                   .Skip(1)
+                                   .Where(g => g.ControlPoints.Any(cp => cp.GetType() == selectedPointType));
```

Comment:
This is the only line that could be removed if the simplification you were hoping for was applied, as far as I can tell.

Reply:
I've read this method in its entirety, then removed this entire code block, and wrote this:

```csharp
if (selectedPointType != null)
{
    // We don't have an efficient way of looking up groups currently, only individual point types.
    // To improve the efficiency of this in the future, we should reconsider the overall structure of ControlPointInfo.

    // Find the next group which has the same type as the selected one.
    selectedGroup.Value = Beatmap.ControlPointInfo.Groups
                                 .Where(g => g.ControlPoints.Any(cp => cp.GetType() == selectedPointType))
                                 .Last(g => g.Time <= clock.CurrentTimeAccurate);
}
```

The above passes tests and seems to be behaving in the exact same way - am I missing something?

Reply:
Isn't that only handling the forward direction?

Reply:
That said, if you prefer the code looking like this I can make two separate linq statements rather than combining both into one.

Reply:
> Isn't that only handling the forward direction?

Not sure I follow. I assume the expectation is that regardless of the direction of seek, the control point selected should be the one that is both (a) in effect and (b) of the same type at the point in time, which is essentially what I wrote. I'm not really sure where directionality comes into it...

Reply:
Fair enough, I was structuring things to be relative to the current control group (with the assumption that it would be optimised in the future or something) but was clearly overthinking. What you have should work fine.